Prep/Source/MeshTools/ParticleBasedSurfaceReconstruction/source/ReconstructSamplesAlongPCAModes.cxx

```diff
@@ -130,17 +143,28 @@ int DoIt(InputParams params)
     {
```

Comment:
This loop may fail when we provide the particular mode index, unless the mode to be generated is very first mode, since NumberOfModes is set to 1.

Reply:
fixed

Reply:
I don't think it is fixed yet, coz the NumberOfModes is set to 1, so loop runs only one time. The continue condition was definitely needed, but the NumberOfModes need to be set to max_possible_number_of_modes. Since we already have the flag for singleModeToBeGen, number of modes need not be set to 1.